src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -478,7 +478,12 @@ void DiscoveryImplPlatform::HandleNodeIdResolve(void * context, MdnsService * re
     nodeData.mAddress     = result->mAddress.ValueOr({});
     nodeData.mPort        = result->mPort;
 
-    ChipLogProgress(Discovery, "Node ID resolved for 0x" ChipLogFormatX64, ChipLogValueX64(nodeData.mPeerId.GetNodeId()));
+#if CHIP_PROGRESS_LOGGING
+    char addrBuffer[INET6_ADDRSTRLEN + 1];
+    nodeData.mAddress.ToString(addrBuffer);
+#endif // CHIP_PROGRESS_LOGGING
```

Comment:
Ifdef extension would make it clearer that addrBuffer definition and usage are one unit.

Reply:
I was actually torn on this.  The reason for the decision I made: If the logging level of the actual log statement changes, if it's included in the #if then it might end up not being compiled when it should be.  This way it will be, and build failures would catch the need to change the #if condition.

Thoughts @andy31415 ?